src/morphology.cpp

```diff
@@ -63,3 +66,39 @@ void section_geometry::segment(double dx) {
     points = std::move(sampled);
     length = sampled_length;
 }
+
+void section_geometry::compute_length() {
```

Comment:
This section_geometry type is getting a bit "big" for the simple struct semantics.

It isn't immediately apparent that `length` is a member of `section_point`. And I am uncomfortable taht it is the user's responsibility to call `compute_length()`.

It might be time to turn this into a proper class.